enhancements/network/bgp-ovn-kubernetes.md

```diff
@@ -0,0 +1,689 @@
+* Configure BFD
+
+FRR-K8S will serve as the main API for user configuration of BGP via FRR-K8S CRs. OVN-Kubernetes will watch for these
+CRs, and once created, OVN-Kubernetes will start monitoring the kernel routing table via netlink for routes installed
+by FRR. When FRR is using Zebra, the routes can be identified via a proto "bgp" label:
+
+```
+123.123.123.0/24 nhid 47 via 172.18.0.5 dev breth0 proto bgp metric 20
+```
+
+When these routes are seen on the node, ovnkube-controller will update the routes in NBDB for any GW routers on the
+node, filtering the route for the GW router based on if the interface on the route matches the interface in OVS. This
+will happen irrespective of gateway mode, as some features in local gateway mode rely on the GRs for sending traffic
+(Egress IP, MEG). Note, this configuration will only happen on nodes selected by the FRR-K8S CRs.
+
+#### API Changes
+
+##### Route Advertisements
+
+When OVN-Kubernetes detects that a FRR-K8S CR has been created for BGP peering, OVN-Kubernetes will by default
+advertise the pod subnet for each node, via creating an additive, per node FRR-K8S CR that is managed by OVN-Kubernetes.
+OVN-Kubernetes CRD:
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: RouteAdvertisements
+metadata:
+  name: default
+spec:
+    # networkSelector:
+    # nodeSelector:
+    # frrConfigurationSelector:
+    targetVRF: default
+    advertisements:
+      podNetwork: true
+      egressIP: true
+      clusterIPs: true
+      externalIPs: true
+```      
+
+In the above example, an optional networkSelector may also be optionally supplied which will match namespaces/networks.
+When omitted, RouteAdvertisements will be applied to the default cluster network. If a selector is used that selects the
+default cluster network, it will be enabled for any namespace using the cluster default network. Additionally,
+advertisements may be limited to specific nodes via the nodeSelector.
+
+A networkSelector may select more than one network, including user-defined networks. In such a case the network subnets
+will be checked by OVN-Kubernetes to determine if there is any overlap of the IP subnets. If so, an error status will be
+reported to the CRD and no BGP configuration will be done by OVN-Kubernetes.
+
+The CRD will support enabling advertisements for pod subnet, egress IPs, as well as cluster IPs and external IPs for
+services on the selected networks. Note, MetalLB only handles advertising LoadBalancer IP so there is no conflict of
+responsibilities here. When the pod network is set to be advertised, there is no longer a need to SNAT pod IPs for this
+network to the node IP. Therefore when pod network advertisements are enabled, the traffic from these pods will no longer
+be SNAT'ed on egress.
+
+The "targetVRF" key is used to determine which VRF the routes should be advertised in. The default value is "auto", in
```

Comment:
Unclear if the default is really default or auto.

Reply:
The default value is "auto". I'm not sure how to reword it to make it more clear. Suggestions? Also, let me know if you think it makes more sense to make the default value "default". I went back and forth on it a few times.

Reply:
In the first example below you are making it behave like if the default was "default" instead of "auto" because targetVFR is omitted and route are being leaked into the default VRF, if I understood things correctly.

Also, having having a valid value as "default" which is not actually the default seems confusing.
Could we make it "default" to be the default?

Reply:
sure I will make that change.